vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -108,7 +116,9 @@ import { extractDateParts, parseDate as _parseDate } from '@vaadin/date-picker/s
           }
 
           function doParseDate(dateString, format, referenceDate) {
-            const date = dateFnsParse(dateString, format, referenceDate);
+            // When format does not contain a year, then current year should be used.
+            const refDate = isFormatWithoutYears(format) ? new Date() : referenceDate;
+            const date = dateFnsParse(dateString, format, refDate);
```

Comment:
This is the actual fix. We could also try `date.setFullYear(new Date().getFullYear())` as an alternative.